### PR TITLE
[WIP] add smoke tests to run in travis 

### PR DESCRIPTION
**Pull request by kimberlythegeek**
added tests to check default config options are set as expected, and custom values are set as expectede

### CODE REVIEW

**Comment by kimberlythegeek** [comment]
yes 29 commits is a lot! I will like do a squash + merge

also I realized that I branched this off of my docker build, I may change that later, but for now mostly looking for a review on the changes and the travis builds

File: moziris/api/settings.py [hunk 1]
```diff
@@ -74,7 +75,7 @@ class _Settings:
     DEFAULT_UI_DELAY_SHORT = 0.5
     DEFAULT_UI_DELAY_LONG = 2.5
     DEFAULT_SYSTEM_DELAY = 5
-    PACKAGE_ROOT = os.path.realpath(os.path.split(__file__)[0] + "/../..")
+    PACKAGE_ROOT = os.getenv("IRIS_CODE_ROOT")
```
Comment (mwobensmith) [inline moziris/api/settings.py]: This isn't right. Package root is not necessarily equal to code root. Package root refers to the relative path from this file, guaranteeing a path to the package. Code root will depend on how this library is deployed, and it's likely different. I'd revert this line back to the original.

File: setup.py [hunk 2]
```diff
@@ -34,11 +34,9 @@
     "pytesseract==0.3.0",
     "pytest==5.1.2",
     "python-dateutil==2.8.0",
+    "python-xlib==0.25",
 ]
 
-if platform.system() == "Linux":
-    INSTALL_REQUIRES.append("xlib")
-
```
Comment (mwobensmith) [inline setup.py]: From my understanding, this conditional import was necessary before to avoid a runtime exception for `xlib` on non-Linux systems. Will your code work cross-platform without raising an exception?

File: setup.py [hunk 3]
```diff
@@ -72,15 +70,10 @@
...
     install_requires=INSTALL_REQUIRES,
     tests_require=TESTS_REQUIRE,
     extras_require={"dev": DEV_REQUIRES},  # For `pip install -e .[dev]`
-    entry_points={
-        "console_scripts": [
-            "iris = moziris.scripts.main:main",
-            "api-test = moziris.scripts.test:api_test",
-        ]
-    },
+    entry_points={"console_scripts": ["iris = moziris.scripts.main:main"]},
```
Comment (mwobensmith) [inline setup.py]: Why did you remove the entry point for `api-test`? We'd like that in there.

File: setup.py (hunk 2 above)
Comment (kimberlythegeek) [reply setup.py]: this may have been the result of tinkering that I missed when committing

**Comment by kimberlythegeek** [comment]
closing in favor of  #639
